overlay/overrides.nix

```diff
@@ -0,0 +1,120 @@
+{ rustLib, lib, buildPackages }:
+let
+  inherit (rustLib) makeOverride nullOverride;
+  envize = s: builtins.replaceStrings ["-"] ["_"] (lib.toUpper s);
+
+  patchOpenssl = pkgs: (pkgs.openssl.override {
+    # `perl` is only used at build time, but the derivation incorrectly uses host `perl` as an input.
+    perl = pkgs.buildPackages.buildPackages.perl;
+  }).overrideAttrs (_: {
+    installTargets = "install_sw";
+    outputs = [ "dev" "out" "bin" ];
+  });
+
+  joinOpenssl = openssl: buildPackages.symlinkJoin {
+    name = "openssl"; paths = with openssl; [ out dev ];
+  };
+
+  patchPostgresql = pkgs: (pkgs.postgresql.override {
+    openssl = patchOpenssl pkgs;
+  }).overrideAttrs (drv: {
+    # We don't need `systemd`. It breaks cross compilation.
+    buildInputs = builtins.filter (d: !lib.hasPrefix "systemd" d.name) drv.buildInputs;
+    configureFlags = builtins.filter (flag: flag != "--with-systemd") drv.configureFlags;
+  });
+
+
+  patchCurl = pkgs:
+    let
+      openssl = patchOpenssl pkgs;
+    in pkgs.curl.override {
+      inherit openssl;
+      nghttp2 = pkgs.nghttp2.override { inherit openssl; };
+      libssh2 = pkgs.libssh2.override { inherit openssl; };
+      libkrb5 = pkgs.libkrb5.override { inherit openssl; };
+    };
+
+in
+rec {
+  patches = [ patchOpenssl patchCurl ];
+
+  # Don't forget to add your new overrides here.
```

Comment:
Might make sense to separate this out into an `all-overrides.nix` file a la Nixpkgs `all-packages.nix` style

Reply:
Overrides are much smaller than packages so I think we can put everything in one file for now.